File: src/main/java/ru/sonyabeldy/historicaldances/dto/DanceListDTO.java
package ru.sonyabeldy.historicaldances.dto;

import lombok.*;
import ru.sonyabeldy.historicaldances.models.Dance;

import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DanceListDTO {

    private int id;
    private String name;
    private Timestamp date;
    private String description;
    private DanceDTO[] dances;
}

File: src/main/resources/static/scripts/dto/dance-list-dto.js
class DanceListDto {
    constructor(id, name, date, desc) {
        this._id = id;
        this._name = name;
        this._date = date;
        this._desc = desc;
    }

    get id() {
        return this._id;
    }

    get name() {
        return this._name;
    }

    get date() {
        return this._date;
    }

    get desc() {
        return this._desc;
    }
}